File: spring-data-microservice-transaction-sample/order-service/src/main/java/sample/order/domain/model/Statement.java
package sample.order.domain.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Table("statements")
public class Statement {

  // This model is actually accessed via a multi-column index, but Spring Data doesn't support it
  // while @Id is always required. So, this @Id annotation is a dummy
  @Id
  public final int itemId;
  public final String orderId;
  public final int count;

  public Statement(int itemId, String orderId, int count) {
    this.itemId = itemId;
    this.orderId = orderId;
    this.count = count;
  }
}

File: dotnet/microservice-transactions-sample-with-shared-cluster-with-linq/Common/OrderService/Statement.cs
using System.ComponentModel.DataAnnotations.Schema;
using ScalarDB.Client.DataAnnotations;

namespace MicroserviceTransactionsSample.Common.OrderService;

[Table("order_service.statements")]
public class Statement
{
    [PartitionKey]
    [Column("order_id", Order = 0)]
    public string OrderId { get; set; } = "";

    [ClusteringKey]
    [Column("item_id", Order = 1)]
    public int ItemId { get; set; }

    [Column("count", Order = 2)]
    public int Count { get; set; }
}
